### PR TITLE
WT-8119 Add targeted compact testing to the existing testing framework

### PR DESCRIPTION
**Pull request by hbokhari**
I am adding targeted compact testing as follows:

1 - Add a new format test config that will exercise compact operation more often
2 - Add compact operation to the existing random_abort csuite test.

### CODE REVIEW

File: test/csuite/random_abort/main.c [hunk 1]
```diff
@@ -217,6 +218,17 @@ thread_run(void *arg)
         if (fprintf(fp[INSERT_RECORD_FILE_ID], "%" PRIu64 "\n", i) == -1)
             testutil_die(errno, "fprintf");
 
+        /*
+         * If configured, run compaction on database after each epoch of 100000 operations.
+         */
+        if (i >= 100000 && i % 100000 == 0 && compaction) {
```
Comment (sueloverso) [inline test/csuite/random_abort/main.c:224]: Minor suggestion: Test `compaction` first in the conditional. No need to perform expensive modulo operations on every iteration for every thread if compaction is turned off.

File: test/csuite/random_abort/main.c [hunk 2]
```diff
@@ -607,18 +619,21 @@ main(int argc, char *argv[])
 
     (void)testutil_set_progname(argv);
 
-    compat = inmem = false;
+    compat = compaction = inmem = false;
```
Comment (sueloverso) [inline test/csuite/random_abort/main.c:622]: Alphabetize in the line

File: test/csuite/random_abort/main.c [hunk 3]
```diff
@@ -607,18 +619,21 @@ main(int argc, char *argv[])
...
     nth = MIN_TH;
     rand_th = rand_time = true;
     timeout = MIN_TIME;
     verify_only = false;
     working_dir = "WT_TEST.random-abort";
 
-    while ((ch = __wt_getopt(progname, argc, argv, "Ch:mT:t:v")) != EOF)
+    while ((ch = __wt_getopt(progname, argc, argv, "cCh:mT:t:v")) != EOF)
```
Comment (sueloverso) [inline test/csuite/random_abort/main.c:629]: Alphabetize `c` after `C`.

**Review by sueloverso — APPROVED** [review]
LGTM with a few minor changes.

**Comment by raviprakashgiri29** [comment]
@hbokhari Can we print the file size before and after compaction just to make sure the test compacts the file?

**Comment by hbokhari** [comment]
@raviprakashgiri29 I look into printing out the file size but the information will be confusing. There are multiple thread inserting at the same time and some of them might be calling compaction in parallel. As such, by the time we collect information about compaction, some other threads might have already inserted the new data.

**Comment by raviprakashgiri29** [comment]
> @raviprakashgiri29 I looked into printing out the file size but the information will be confusing. There are multiple thread inserting at the same time and some of them might be calling compaction in parallel. As such, by the time we collect information about compaction, some other threads might have already inserted the new data.

Yeah, it makes sense now. Thanks, @hbokhari for the clarification.

**Review by raviprakashgiri29 — APPROVED** [review]
LGTM

File: test/csuite/random_abort/main.c (hunk 1 above)
Comment (hbokhari) [reply test/csuite/random_abort/main.c]: Fixed.

File: test/csuite/random_abort/main.c (hunk 2 above)
Comment (hbokhari) [reply test/csuite/random_abort/main.c]: Fixed.

File: test/csuite/random_abort/main.c (hunk 3 above)
Comment (hbokhari) [reply test/csuite/random_abort/main.c]: Fixed.

**Pull request merged**